.github/workflows/build.yml

```diff
@@ -76,43 +76,29 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - uses: actions/setup-python@v4
-        name: Set up Python
-        with:
-          python-version: ${{ matrix.python-version }}
-
-      - name: Install dependencies
+      - name: Workaround directory permissions
         run: |
-          python -m pip install --upgrade pip
-          pip install coverage
-
-      - name: Export PATH
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-
-      - uses: actions/checkout@v3
-        with:
-          repository: bats-core/bats-core
-          ref: v1.7.0
-          path: test/bats-core
-      - run: test/bats-core/install.sh $HOME/.local
-
-      - name: Use git-pile from checkout
-        run: echo "$PWD" >> $GITHUB_PATH
+          chown -R $(id -u):$(id -g) .
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
 
+      # should be supported in any version of git we support
       - name: Setup git
         run: |
           git config --global user.name "ci"
           git config --global user.email "ci@git-pile.github.io"
           git config --global init.defaultBranch master
 
-      # User in the container will be root, but the cloned repo is owned by the
-      # user running github-runner. Git doesn't like that as the repo will then
-      # be unusable by the owner:  we don't care about it since this is only
-      # used for running these tests.
-      #
-      # See https://github.com/actions/runner/issues/2033
-      - name: Workaround directory permissions
-        run: git config --global --add safe.directory $PWD
+      - uses: actions/setup-python@v4
+        name: Set up Python
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --user --upgrade pip coverage
```

Comment:
Couldn't this be covered by `python3 -m pip install --user --no-use-pep517 --editable .[tests]` in the step below?

Reply:
The dot (".") is the directory, just like it is already being done. Each name inside the brackets refer to the group of optional dependencies declared in the `extras_require` from `setup.py`.

Reply:
Didn't notice this was also installing `pip`. Anyways, either this step could be kept (installing only pip) or "merged" with the one below. In any case, I still think it would be nice to have dependencies for test being installed together with git-pile with ".[tests]".

Reply:
done